gui/src/controls/control.ts

```diff
@@ -768,6 +768,13 @@ module BABYLON.GUI {
         
             this._fontOffset = Control._GetFontOffset(this._font);
         }
+        
+        public measureForIdeal() {
+            this._measure();
+            
+            this.width = this._currentMeasure.width + 'px';

```

Comment:
For some reason in PG _currentMeasure returns 0, 0 for me, while locally it works - I probably created locally a situations where it works completely by luck, as like I said I don't entirely understand what's going on in the source (well, far from it).

Anyway, I hope this PG could at least help you understand my goal:
http://www.babylonjs-playground.com/#XCPP9Y#66